command/harness/dlite/gitspaces_cleanup.go

```diff
@@ -0,0 +1,67 @@
+package dlite
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/drone-runners/drone-runner-aws/command/harness"
+	"github.com/sirupsen/logrus"
+	"github.com/wings-software/dlite/client"
+	"github.com/wings-software/dlite/httphelper"
+)
+
+type GitspacesVMCleanupTask struct {
+	c *dliteCommand
+}
+
+func (t *GitspacesVMCleanupTask) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithCancel(r.Context()) // TODO: Get this from http Request
+	defer cancel()
+	log := logrus.New()
+	task := &client.Task{}
+	err := json.NewDecoder(r.Body).Decode(task)
+	if err != nil {
+		log.WithError(err).Error("could not decode VM cleanup HTTP body")
+		httphelper.WriteBadRequest(w, err)
+		return
+	}
+	logr := log.WithField("task_id", task.ID)
+	// Unmarshal the task data
+	taskBytes, err := task.Data.MarshalJSON()
+	if err != nil {
+		logr.WithError(err).Error("could not unmarshal task data")
+		httphelper.WriteBadRequest(w, err)
+		return
+	}
+	req := &harness.VMCleanupRequest{}
+	err = json.Unmarshal(taskBytes, req)
+	if err != nil {
+		logr.WithError(err).Error("could not unmarshal task request data")
+		httphelper.WriteBadRequest(w, err)
+		return
+	}
+	accountID := harness.GetAccountID(&req.Context, map[string]string{})
+	poolManager := t.c.getPoolManager(req.Distributed)
+	if !req.Distributed {
+		harness.GetCtxState().Delete(req.StageRuntimeID)
+	}
+	err = harness.HandleDestroy(ctx, req, poolManager.GetStageOwnerStore(), &t.c.env, poolManager, t.c.metrics)
+	if err != nil {
+		t.c.metrics.ErrorCount.WithLabelValues(accountID, strconv.FormatBool(req.Distributed)).Inc()
+		logr.WithError(err).WithField("account_id", accountID).Error("could not destroy VM")
+		httphelper.WriteJSON(w, failedResponse(err.Error()), httpFailed)
+		return
+	}
+	resp := GitspacesVMTaskExecutionResponse{
+		VMTaskExecutionResponse: VMTaskExecutionResponse{
+			CommandExecutionStatus: Success,
+			DelegateMetaInfo: DelegateMetaInfo{
+				HostName: t.c.delegateInfo.Host,
+				ID:       t.c.delegateInfo.ID,
+			},
+		},
+	}
+	httphelper.WriteJSON(w, resp, httpOK)
```

Comment:
It would be easier to understand if we break the code in blocks eg

```
accountID := harness.GetAccountID(&req.Context, map[string]string{})

poolManager := t.c.getPoolManager(req.Distributed)
	
if !req.Distributed {
	harness.GetCtxState().Delete(req.StageRuntimeID)
}
	
err = harness.HandleDestroy(ctx, req, poolManager.GetStageOwnerStore(), &t.c.env, poolManager, t.c.metrics)
if err != nil {
	t.c.metrics.ErrorCount.WithLabelValues(accountID, strconv.FormatBool(req.Distributed)).Inc()
	
         logr.WithError(err).WithField("account_id", accountID).Error("could not destroy VM")
	
         httphelper.WriteJSON(w, failedResponse(err.Error()), httpFailed)
		
        return
}
```

Reply:
Most of the code in gitspaces_cleanup.go is borrowed from CI's cleanup.go. Same with gitspaces_init.go and init.go.
We have only changed the necessary things. 
gitspaces_init.go -> Request, Response, the additional parameter passed to HandleSetup method
gitspaces_cleanup.go -> Response

Reply:
IMO We can move duplicate things to common code and make it more modular later. And that's when we can take up all the renaming items.
For now the plan is to make it work while not causing any regression in CI

Reply:
yeah lets duplicate till we have complete flow, working and then come to refactoring, we don't have bandwidth to handle regression in existing flow.

Reply:
sounds good @vikyathharekal

Reply:
I would push for removing duplication in this PR itself unless there's a very strong urgency, otherwise changes can very easily be missed for gitspaces.

Also I see two approaches here - one for adding gitspaces config in VM init/exec flows and the other (which you have done) is to create a separate task  and add in the new config required to those task structs. IMO the first approach would lead to less duplication and you could even keep the handler codes same. We could accept a gitspaces config in `SetupVMRequest` and logically speaking setting up gitspaces is akin to setting up a VM with some extra config. Was there any reason why you went with the second approach instead?

Reply:
@vikyathharekal @n00bitax @harshjain12 ^